modules/images/dominant-color/load.php

```diff
@@ -0,0 +1,425 @@
+
+		if ( ! empty( $dominant_color ) ) {
+			$metadata['dominant_color'] = $dominant_color;
+		}
+
+		return $metadata;
+	}
+
+	/**
+	 * Add the dominant color metadata to the attachment.
+	 *
+	 * @param array $metadata
+	 * @param int $attachment_id
+	 *
+	 * @return array $metadata
+	 */
+	public function has_transparency_metadata( $metadata, $attachment_id ) {
+
+		$has_transparency = $this->get_has_transparency( $attachment_id );
+
+		if ( ! empty( $has_transparency ) ) {
+			$metadata['has_transparency'] = $has_transparency;
+		}
+
+		return $metadata;
+	}
+
+	/**
+	 * @param $filtered_image
+	 * @param $context
+	 * @param $attachment_id
+	 *
+	 * @return string image tag
+	 */
+	public function tag_add_adjust( $filtered_image, $context, $attachment_id ) {
+
+		$image_meta = wp_get_attachment_metadata( $attachment_id );
+
+		if ( isset( $image_meta['dominant_color'] ) ) {
+			$data  = sprintf( 'data-dominantColor="%s"', $image_meta['dominant_color'] );
+			$style = '';
+			if ( strpos( $filtered_image, 'loading="lazy"' ) !== false && ( isset( $image_meta['show_dominant_color'] ) && 1 === (int) $image_meta['show_dominant_color'] ) ) {
+				$style = ' style="--dominant-color: #' . $image_meta['dominant_color'] . '; " ';
+			}
+
+			$extra_class = '';
+
+			if ( ! isset( $image_meta['has_transparency'] ) ) {
+				$data        .= ' data-hasTransparency="false"';
+			} else {
+				$data .= ' data-hasTransparency="true"';
+				$extra_class = ' has-transparency ';
+			}
+
+			$filtered_image = str_replace( '<img ', '<img ' . $data . $style, $filtered_image );
+
+			$extra_class    .= ( $this->colorislight( $image_meta['dominant_color'] ) ) ? 'dominant-color-light' : 'dominant-color-dark';
+			$filtered_image = str_replace( 'class="', 'class="' . $extra_class . ' ', $filtered_image );
+		}
+
+		return $filtered_image;
+	}
+
+
+	/**
+	 * @return void
+	 */
+	public function add_styles() {
+		?>
+		<style>
+			img[data-dominantcolor]:not(.has-transparency) {
+				background-color: var(--dominant-color);
+				background-clip: content-box, padding-box;
+			}
+		</style>
+		<?php
+	}
+
+	public function add_scripts() {
+		?>
+<script>
+	(function (window, document) {
+		'use strict';
+		document.querySelector('img[data-dominantcolor]').addEventListener('load', (event) => {
+			var image = event.target;
+			image.style.setProperty('--dominant-color', '');
+			});
+	})(window, document);
+</script>
+		<?php
+	}
+
+
+
+	/**
+	 * @return void
+	 */
+//	public function enqueue_plugin_scripts() {
+//		wp_enqueue_script( 'blurhash', plugin_dir_url( __FILE__ ) . 'dist/blurhash.js' );
+//	}
+
+	/**
+	 * @param $content
+	 * @param $context
+	 *
+	 * @return string content
+	 */
+	public function filter_content_tags( $content, $context = null ) {
+		if ( null === $context ) {
+			$context = current_filter();
+		}
+
+		if ( ! preg_match_all( '/<(img|iframe)\s[^>]+>/', $content, $matches, PREG_SET_ORDER ) ) {
```

Comment:
Why is this searching for iframes?
It should probably be just img tags.

Reply:
this is a direct copy from  core 
I can remove this or keep so we can keep parity with core

Reply:
I would at the very least remove the |iframe from it. we need to bail as soon as we don't have an image to work with.